assets/js/components/ProjectEditorTextField.js

```diff
@@ -0,0 +1,163 @@
+  this.translationDeleted = projectDescriptionCredits.data('trans-translation-deleted')
+
+  this.pathEditName = projectDescriptionCredits.data('path-edit-program-name')
+  this.pathEditDescription = projectDescriptionCredits.data('path-edit-program-description')
+  this.pathEditCredits = projectDescriptionCredits.data('path-edit-program-credits')
+
+  this.show = () => {
+    this.editText.val(this.initialText)
+    this.lastSavedText = this.initialText
+  }
+
+  this.close = () => {
+    self.editTextError.hide()
+  }
+
+  this.areChangesSaved = () => {
+    return this.editText.val().trim() === this.lastSavedText.trim()
+  }
+
+  this.save = (languageSelected) => {
+    const newText = this.editText.val().trim()
+    if (this.areChangesSaved()) {
+      this.close()
+      return
+    }
+
+    if (!this.hasDefaultText && !(languageSelected === '' || languageSelected === 'default')) {
+      return
+    }
+
+    if (languageSelected === '' || languageSelected === 'default') {
+      let url
+      if (this.programSection === 'name') {
+        url = this.pathEditName
+      } else if (this.programSection === 'description') {
+        url = this.pathEditDescription
+      } else if (this.programSection === 'credits') {
+        url = this.pathEditCredits
+      }
+
+      return new Promise(function (resolve, reject) {
+        $.ajax({
+          url: url,
+          type: 'put',
+          data: { value: newText },
+          success: function (data) {
+            const statusCode = parseInt(data.statusCode)
+            if (statusCode === 527 || statusCode === 707) {
+              showError(data)
+              reject(statusCode)
+            } else {
+              resolve(statusCode)
+            }
+          },
+          error: function (error) {
+            if (error.status === 422) {
+              error.message = error.responseText
+              showError(error)
+              reject(error.status)
+            } else if (error.status === 401) {
+              reject(error.status)
+            }
+          }
+        })
+      })
+    } else if (newText === '') {
+      return this.customTranslationApi.deleteCustomTranslation(
+        this.programId,
+        languageSelected,
+        deleteCustomTranslationSuccess,
+        showError
+      )
+    } else {
+      return this.customTranslationApi.saveCustomTranslation(
+        this.programId,
+        newText,
+        languageSelected,
+        saveCustomTranslationSuccess,
+        showError
+      )
+    }
+  }
+
+  function showError (error) {
+    self.editText.addClass('danger')
+    self.editTextError.show()
+    self.editTextError.text(error.message)
+  }
+
+  function deleteCustomTranslationSuccess (language) {
+    self.snackbar.show(self.translationDeleted, language)
+    self.close()
+  }
+
+  function saveCustomTranslationSuccess (language) {
+    self.snackbar.show(self.translationSaved, language)
```

Comment:
so this screen will have 3 snackbars? I think we could use only one.

Reply:
Refactored this logic to only show one snackbar for save and one for delete